cmake/unity.cmake

```diff
@@ -0,0 +1,12 @@
+function(TT_ENABLE_UNITY_BUILD TARGET)
+    if(TT_UNITY_BUILDS)
+        set_target_properties(
+            ${TARGET}
+            PROPERTIES
+                UNITY_BUILD
+                    ON
+                UNITY_BUILD_UNIQUE_ID
```

Comment:
Let's put this in `cmake/unity.cmake`.  I can't think of any scenario where we DON'T want it, nor where we want different define names.  So it makes sense to leverage the global setting and leave the per-target config to what needs to be tweaked per-target.